presto-spi/src/main/java/io/prestosql/spi/type/DoubleType.java

```diff
@@ -158,4 +152,12 @@ public int hashCode()
     {
         return getClass().hashCode();
     }
+
+    @Override
+    public Optional<Range> getRange()
+    {
+        // The range for double is undefined because NaN is a special value that
+        // is *not* in any reasonable definition of a range for this type.
```

Comment:
NaN cannot be produced by a cast from exact numeric to a floating point.
So for these casts is as if NaN value did not exist, and the rule could just work for `*int* → double` casts.
If the literal is NaN... this may require special handling.

I am not insisting any change now, but it seems there might be room for improvement.

Reply:
The rule *does* work for int to double already. A range is only needed for casts *from* the type (and not really need, just an additional optimization)